core/src/main/java/org/jdbi/v3/core/PreparedBatch.java

```diff
@@ -60,16 +59,27 @@
         this.currentBinding = new Binding();
     }
 
+    @Override
+    public <R> R withResultSet(ResultSetCallback<R> callback) {
+        return execute(returningResults()).withResultSet(callback);
+    }
+
     /**
      * Execute the batch
      *
      * @return the number of rows modified or inserted per batch part.
      */
     public int[] execute() {
-        return (int[]) internalBatchExecute(null, null);
+        // short circuit empty batch
+        if (parts.isEmpty()) {
```

Comment:
I'm going to go ahead and merge this. If you feel strongly about the empty behavior, then let's discuss it in a follow-up ticket.